hathor/verification/token_creation_transaction_verifier.py

```diff
@@ -39,20 +40,14 @@ def verify_sum(self, tx: Transaction) -> None:
         :raises InputOutputMismatch: if sum of inputs is not equal to outputs and there's no mint/melt
         """
         assert isinstance(tx, TokenCreationTransaction)
-        token_dict = tx.get_token_info_from_inputs()
-
-        # we add the created token's info to token_dict, as the creation tx allows for mint/melt
-        assert tx.hash is not None
-        token_dict[tx.hash] = TokenInfo(0, True, True)
-
-        self.update_token_info_from_outputs(tx, token_dict=token_dict)
+        token_dict = self.get_complete_token_info(tx)
 
         # make sure tokens are being minted
-        token_info = token_dict[tx.hash]
+        token_info = token_dict[not_none(tx.hash)]
         if token_info.amount <= 0:
             raise InvalidToken('Token creation transaction must mint new tokens')
 
-        self.verify_authorities_and_deposit(token_dict)
+        super().verify_sum(tx)
```

Comment:
I guess we are calculating `token_info` twice if we change this call to `super().verify_sum(tx)`, right? In this case, it would also double the number of hits in the storage to load the inputs.

Reply:
That's true, but it would most likely hit the cache. Do you think it could be a problem?

If yes, an alternative is changing the code so every dependency is fetched before the verification process starts, we've talked about this change. For example, we would fetch all inputs and parents, and provide them in a dict to the verifier methods, so no method would need to call the storage. The calculated `token_info` could be in this dependencies object, too. This change will also be useful for the future project of running verifications on a separate process, as it won't have to access the storage in parallel.

If considering the cache this is not problem, I would rather do this after the current refactors. If it is, I can do it now in a separate PR. What do you think?

Reply:
I guess we should do it now in a separate PR because vertex validation is the bottleneck and I feel we shouldn't calculate the same thing more than once.

Reply:
Ok, makes sense! I'll move this PR back to WIP.

Reply:
This has been implemented in 2 PRs, that depend on this one. Here they are: https://github.com/HathorNetwork/hathor-core/pull/858 and https://github.com/HathorNetwork/hathor-core/pull/859. Let's review all 3 of them before merging this one, so they call all be merged together.

Reply:
Considering what was discussed in a meeting, disregard the comment above. This has been fixed in a new PR: https://github.com/HathorNetwork/hathor-core/pull/868

Reply:
Ok. Thanks!